app/routines/newflow/activate_user.rb

```diff
@@ -1,12 +1,16 @@
+# Sets the passed-in user's `state` to `'activated'`
+# and pushes the user to salesforce as a new lead
+# If the user is already `activated`, then it does nothing.
 module Newflow
   class ActivateUser
     lev_routine
 
     protected ###############
 
     def exec(user)
-      push_lead_to_salesforce(user)
+      return if user.is_activated?
```

Comment:
is_X? (is and ?) isnt idiomatic ruby.   that's a java thing.   suggestion...change this to `user.activated?`